be/src/olap/rowset/segment_creator.cpp

```diff
@@ -344,7 +342,7 @@ Status SegmentFlusher::_flush_segment_writer(
 
     writer.reset();
 
-    RETURN_IF_ERROR(_context->segment_collector->add(segment_id, segstat));
+    RETURN_IF_ERROR(_context->segment_collector->add(segment_id, segstat, _flush_schema));
```

Comment:
Add an argument `FlushSchemaSPtr flush_schema` to `_flush_segment_writer()`, avoid adding `_flush_schema` to the class.

Reply:
done